### PR TITLE
Warning for VSO errors

### LINKED ISSUE #3338: Return VSO errors to the user
Currently we hide any VSO error codes from the user at Search time. If there is a reliable way we can extract this info we should throw warnings or similar.

From #3331 we can see that under certain conditions we get an error key in the returned XML from the VSO:
```
DEBUG: VSO Response:
 <soap:Envelope xmlns:VSO="http://virtualsolar.org/VSO/VSOi" xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
  <soap:Body>
    <VSO:QueryResponse>
      <body>
        <provideritem>
          <error>VSO-D404 Bad Request -- Invalid Time Sample : contains non digit</error>
          <provider>SSC</provider>
          <version>1</version>
        </provideritem>
      </body>
    </VSO:QueryResponse>
  </soap:Body>
</soap:Envelope>
```

The question here is, is this layout consistent over all providers and error types the VSO might throw back at us, so should we be parsing this and raising it as a warning during `VSOClient.search`.

### PR DESCRIPTION
**Pull request by jains8844**
Fixes #3338

### CODE REVIEW

**Comment by jains8844** [comment]
For JSOC one, there was some issue. Kindly refer this [https://github.com/sunpy/sunpy/issues/3338#issuecomment-576259172](https://github.com/sunpy/sunpy/issues/3338#issuecomment-576259172)

**Comment by jains8844** [comment]
I formatted the code with black which resulted in so many changes. The main change is just a single line :`) [change](https://github.com/sunpy/sunpy/pull/3719/files#diff-0c4a0061c4c5eefc3ba432fa7a2b1280L367)

**Comment by nabobalis** [comment]
I will have review this PR later on but we can't accept black formatted. I suggest you undo that formatting.

**Comment by jains8844** [comment]
Okay no problem.

**Comment by jains8844** [comment]
Done.

**Comment by nabobalis** [comment]
Thanks.

**Comment by nabobalis** [comment]
Can you construct a test for this? It might be a pain to make but we want to make sure it catches/reports errors how we think it does.

**Comment by jains8844** [comment]
How do I test if some code raises a warning ??

**Comment by nabobalis** [comment]
Using pytest: https://docs.pytest.org/en/latest/assert.html#assertions-about-expected-exceptions

**Comment by jains8844** [comment]
Okay thanks :)

**Comment by jains8844** [comment]
Implemented the test.

**Comment by jains8844** [comment]
Kindly review this

File: sunpy/net/tests/test_fido.py [hunk 1]
```diff
@@ -229,6 +230,17 @@ def test_vso_unifiedresponse():
     assert isinstance(uresp, UnifiedResponse)
 
 
+@pytest.mark.remote_data
+def test_vso_error():
+    with pytest.warns(SunpyUserWarning,
+        match="VSO-D404 Bad Request -- Invalid Time \
+Sample : contains non digit"):
```
Comment (nabobalis) [inline sunpy/net/tests/test_fido.py]: I don't think this line continuation, can you reformat this please?

Comment (jains8844) [reply sunpy/net/tests/test_fido.py]: Done

File: sunpy/net/tests/test_fido.py [hunk 2]
```diff
@@ -229,6 +230,16 @@ def test_vso_unifiedresponse():
     assert isinstance(uresp, UnifiedResponse)
 
 
+@pytest.mark.remote_data
+def test_vso_error():
+    with pytest.warns(SunpyUserWarning,
+        match="VSO-D404 Bad Request -- Invalid Time Sample : contains non digit"):
```
Comment (dstansby) [inline sunpy/net/tests/test_fido.py]: This seems like an odd warning to be raised - the query below looks fine to me. Do you know why it raises this warning?

Comment (Cadair) [reply sunpy/net/tests/test_fido.py]: I think it's https://github.com/sunpy/sunpy/issues/3331

Comment (jains8844) [reply sunpy/net/tests/test_fido.py]: Yes it is #3331

Comment (nabobalis) [reply sunpy/net/tests/test_fido.py]: Do we want to have a test that might get invalidated down the road?

Comment (jains8844) [reply sunpy/net/tests/test_fido.py]: Can you explain a little bit ? I didn't get it

**Comment by jains8844** [comment]
This should be a trivial change right ?

File: sunpy/net/tests/test_fido.py (hunk 2 above)
Comment (nabobalis) [reply sunpy/net/tests/test_fido.py]: The test you wrote is using a bug to raise the warning which could get fixed in the near future.

Comment (jains8844) [reply sunpy/net/tests/test_fido.py]: Oh yeah got it. I couldn't find another query so I used this .

Comment (jains8844) [reply sunpy/net/tests/test_fido.py]: I got one that throws 404 Not Found error. Would that work ?

**Comment by nabobalis** [comment]
Yeah trivial would work.

File: sunpy/net/tests/test_fido.py (hunk 2 above)
Comment (nabobalis) [reply sunpy/net/tests/test_fido.py]: Can you add it in addition to this test? Just add another with block.

File: changelog/3719.trivial.rst [hunk 3]
```diff
@@ -0,0 +1 @@
+Added support to return errors from VSO as warnings in `sunpy.net.vso`
```
Comment (nabobalis) [inline changelog/3719.trivial.rst]:
```suggestion
Errors from a VSO search will now be raised to the user .
```

**Comment by jains8844** [comment]
Commited :)

**Comment by nabobalis** [comment]
Thanks @jains8844

**Pull request merged**